Source/aweXpect.Reflection/ThatMethod.cs

```diff
@@ -1,8 +1,117 @@
-﻿using System.Reflection;
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using aweXpect.Core;
+using aweXpect.Core.Constraints;
+using aweXpect.Reflection.Helpers;
+using aweXpect.Results;
 
 namespace aweXpect.Reflection;
 
 /// <summary>
 ///     Expectations on <see cref="MethodInfo" />.
 /// </summary>
-public static partial class ThatMethod;
+public static partial class ThatMethod
+{
+	/// <summary>
+	///     Verifies that the <see cref="MethodInfo" /> has a parameter of type <typeparamref name="T" />.
+	/// </summary>
+	public static AndOrResult<MethodInfo?, IThat<MethodInfo?>> HasParameter<T>(this IThat<MethodInfo?> subject)
```

Comment:
The return types should be the same as for the filtering (`WithParameter`), so that I can chain them with
1. `AtIndex(int)` / `AtIndex(int).FromEnd()` (consider using the `HasItemResult<TCollection>` from aweXpect.Core as base class)
2. `WithoutDefaultValue()`
3. `WithDefaultValue()` / `WithDefaultValue<T>()` / `WithDefaultValue<T>(Func<T, bool>)`
4. For parameters with expected name, all methods from the `StringEqualityTypeResult` / `StringEqualityResult` should be possible.